wandb/sdk/interface/artifacts.py

```diff
@@ -896,9 +932,9 @@ def cleanup(self, target_size: int) -> int:
             bytes_reclaimed += stat.st_size
         return bytes_reclaimed
 
-    def _cache_opener(self, path):
+    def _cache_opener(self, path: StrPath) -> "Opener":
         @contextlib.contextmanager
-        def helper(mode="w"):
+        def helper(mode: str = "w") -> Generator[IO, None, None]:
```

Comment:
I've always been a little fuzzy on `Iterator` vs `Generator` vs `Coroutine`, but I think this can be simplified to:
```suggestion
        def helper(mode: str = "w") -> Iterator[IO]:
```

Reply:
Usually actual generators (ones that are intended to yield a series of 0 or more values) are best typed as `Iterators`, but when generators are used as context managers they get typed as `_GeneratorContextManager`, and I don't think you can make that work if you type it as Iterator before wrapping it. I'll try it though, the `Generator[X, None, None]` always annoys me.